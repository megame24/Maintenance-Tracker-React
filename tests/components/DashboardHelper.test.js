import React from 'react';
import { shallow, mount } from "enzyme";
import {
  Table, ApproveAndDisapproveColumn, ResolveColumn,
  TitleColumn,
} from '../../src/components/helpers/DashboardHelper';


describe('Testing TitleColumn component', () => {
  const request = {
    title: 'title',
    type: 'repair',
    id: 1,
    status: 'pending',
  };
  it('should render as expected', () => {
    const tree = shallow(<TitleColumn request={request} />);
    expect(tree).toMatchSnapshot();
  });
});

describe('Testing Table component', () => {
  const props = {
      requests: [{
      title: 'title',
      type: 'repair',
      id: 1,
      status: 'pending',
    }],
    approveRequest: () => {},
    disapproveRequest: () => {},
    resolveRequest: () => {},
  };
  it('should render as expected', () => {
    const tree = shallow(<Table {...props} />);
    expect(tree).toMatchSnapshot();
  });
});

describe('Testing ResolveColumn component', () => {
  const props = {
      request: {
      title: 'title',
      type: 'repair',
      id: 1,
      status: 'approved',
    },
    resolveRequest: () => {},
  };
  it('should render as expected when request status is pending', () => {
    const tree = shallow(<ResolveColumn {...props} />);
    expect(tree).toMatchSnapshot();
  });
  it('should render as expected when request status is resolved', () => {
    props.request.status = 'resolved'
    const tree = shallow(<ResolveColumn {...props} />);
    expect(tree).toMatchSnapshot();
  });
});



describe('Testing ApproveAndDisapproveColumn component', () => {
  const props = {
      request: {
      title: 'title',
      type: 'repair',
      id: 1,
      status: 'pending',
    },
    approveRequest: () => {},
    disapproveRequest: () => {},
  };
  it('should render as expected when request status is pending', () => {
    const tree = shallow(<ApproveAndDisapproveColumn {...props} />);
    expect(tree).toMatchSnapshot();
  });
  it('should render as expected when request status is approved', () => {
    props.request.status = 'approved'
    const tree = shallow(<ApproveAndDisapproveColumn {...props} />);
    expect(tree).toMatchSnapshot();
  });
  it('should render as expected when request status is disapproved', () => {
    props.request.status = 'disapproved'
    const tree = shallow(<ApproveAndDisapproveColumn {...props} />);
    expect(tree).toMatchSnapshot();
  });
  it('should render as expected when request status is resolved', () => {
    props.request.status = 'resolved'
    const tree = shallow(<ApproveAndDisapproveColumn {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
