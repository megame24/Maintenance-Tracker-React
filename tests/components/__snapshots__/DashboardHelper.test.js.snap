// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ApproveAndDisapproveColumn component should render as expected when request status is approved 1`] = `
<td>
  Approved
</td>
`;

exports[`Testing ApproveAndDisapproveColumn component should render as expected when request status is disapproved 1`] = `
<td>
  Disapproved
</td>
`;

exports[`Testing ApproveAndDisapproveColumn component should render as expected when request status is pending 1`] = `
<td>
  <span>
    <a
      className="approve btn btn-small btn-primary"
      onClick={[Function]}
    >
      Approve
    </a>
       
    <a
      className="disapprove btn btn-small btn-danger"
      onClick={[Function]}
    >
      Disapprove
    </a>
  </span>
</td>
`;

exports[`Testing ApproveAndDisapproveColumn component should render as expected when request status is resolved 1`] = `
<td>
  Approved
</td>
`;

exports[`Testing ResolveColumn component should render as expected when request status is pending 1`] = `
<td>
  <a
    className="resolve btn btn-small btn-success"
    onClick={[Function]}
  >
    Resolve
  </a>
</td>
`;

exports[`Testing ResolveColumn component should render as expected when request status is resolved 1`] = `
<td>
  Resolved
</td>
`;

exports[`Testing Table component should render as expected 1`] = `
<table>
  <thead>
    <tr>
      <th>
        Title
      </th>
      <th
        className="table-not-mobile"
      >
        Type
      </th>
      <th>
        Approve / Disapprove
      </th>
      <th>
        Resolve
      </th>
    </tr>
  </thead>
  <tbody
    id="table-body"
  >
    <tr
      key="1"
    >
      <TitleColumn
        request={
          Object {
            "id": 1,
            "status": "pending",
            "title": "title",
            "type": "repair",
          }
        }
      />
      <td
        className="table-not-mobile"
      >
        repair
      </td>
      <ApproveAndDisapproveColumn
        approveRequest={[Function]}
        disapproveRequest={[Function]}
        request={
          Object {
            "id": 1,
            "status": "pending",
            "title": "title",
            "type": "repair",
          }
        }
      />
      <ResolveColumn
        request={
          Object {
            "id": 1,
            "status": "pending",
            "title": "title",
            "type": "repair",
          }
        }
        resolveRequest={[Function]}
      />
    </tr>
  </tbody>
</table>
`;

exports[`Testing TitleColumn component should render as expected 1`] = `
<td
  className="dark-blue table-link"
>
  <Link
    replace={false}
    to="/admin/requests/1"
  >
    title
  </Link>
</td>
`;
